### PR TITLE
Fix Joni regexp functions for multi-byte characters and empty pattern

### PR DESCRIPTION
**Pull request by sopel39**

### CODE REVIEW

**Comment by sopel39** [comment]
There is similar issue with `regexpExtractAll`, `regexpSplit` and `JoniRegexpReplaceLambdaFunction`

File: presto-main/src/test/java/io/prestosql/operator/scalar/AbstractTestRegexpFunctions.java [hunk 1]
```diff
@@ -89,6 +89,9 @@ public void testRegexCharLike()
     @Test
     public void testRegexpReplace()
     {
+        assertFunction("REGEXP_REPLACE('abc有朋$%X自9远方来', '', 'Y')", createVarcharType(97), "YaYbYcY有Y朋Y$Y%YXY自Y9Y远Y方Y来Y");
```
Comment (findepi) [inline presto-main/src/test/java/io/prestosql/operator/scalar/AbstractTestRegexpFunctions.java]: Add also other tests for multi-byte.

```
assertFunction("REGEXP_REPLACE('a有朋💰', '.', 'Y')", createVarcharType(4), "YYYY");
```

```
assertFunction("REGEXP_REPLACE('a有朋💰', '.', '1$12')", createVarcharType(..), "1a21有21朋21💰2");
```

**Comment by sopel39** [comment]
fixed other functions too

File: presto-main/src/test/java/io/prestosql/operator/scalar/AbstractTestRegexpFunctions.java [hunk 2]
```diff
@@ -221,6 +238,13 @@ public void testRegexpExtractAll()
     @Test
     public void testRegexpSplit()
     {
+        assertFunction(
+                "REGEXP_SPLIT('abc有朋$%X自9远方来💰', '')",
+                new ArrayType(createVarcharType(14)),
+                ImmutableList.of("", "a", "b", "c", "有", "朋", "$", "%", "X", "自", "9", "远", "方", "来", "💰", ""));
```
Comment (findepi) [inline presto-main/src/test/java/io/prestosql/operator/scalar/AbstractTestRegexpFunctions.java]: i guess this is pre-existing, but i don't underdstand why would split want to return `""` on both ends.

Comment (sopel39) [reply presto-main/src/test/java/io/prestosql/operator/scalar/AbstractTestRegexpFunctions.java]: I guess you match beginning of string, then on the left you have empty string and on the right you have a letter.

Similarly, `"REGEXP_SPLIT('abc', 'a')` would give you: `"", "bc`

**Pull request merged**